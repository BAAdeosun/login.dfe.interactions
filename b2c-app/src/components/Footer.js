import React from 'react'

export default function footer() {
    return (
        <div>
            I AM THE FOOTER
        </div>
    )
}
