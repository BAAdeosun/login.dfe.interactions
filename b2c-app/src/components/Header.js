import React from 'react'

function Header() {
    return (
        <header class="govuk-header " role="banner" data-module="govuk-header">
            <div class="govuk-header__container govuk-width-container">
                <div class="govuk-header__content">
                <a href="#" class="govuk-header__link govuk-header__link--service-name">
                    National Careers Service
                </a>
                <button type="button" class="govuk-header__menu-button govuk-js-header-toggle" aria-controls="navigation" aria-label="Show or hide Top Level Navigation">Menu</button>
                <nav>
                    <ul id="navigation" class="govuk-header__navigation " aria-label="Top Level Navigation">
                    <li class="govuk-header__navigation-item govuk-header__navigation-item--active">
                        <a class="govuk-header__link" href="#1">
                        Job profiles
                        </a>
                    </li>
                    <li class="govuk-header__navigation-item">
                        <a class="govuk-header__link" href="#2">
                        Skills health check
                        </a>
                    </li>
                    <li class="govuk-header__navigation-item">
                        <a class="govuk-header__link" href="#3">
                        Find a course
                        </a>
                    </li>
                    <li class="govuk-header__navigation-item">
                        <a class="govuk-header__link" href="#4">
                        Contact an adviser
                        </a>
                    </li>
                    <li class="govuk-header__navigation-item">
                        <a class="govuk-header__link" href="#4">
                        About us
                        </a>
                    </li>
                    </ul>
                </nav>
                </div>
            </div>
        </header>
    )
}

export default Header;