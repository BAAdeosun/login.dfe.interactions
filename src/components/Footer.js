import Link from 'next/link'

const Footer = () => (
    <footer>
        This is the footer
    </footer>
)

export default Footer