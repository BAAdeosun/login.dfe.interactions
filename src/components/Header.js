import Link from 'next/link'

const Header = () => (
    <header>
        This is the header
    </header>
)

export default Header