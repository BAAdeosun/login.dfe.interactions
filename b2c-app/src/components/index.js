import Header from './Header/Header';
import Footer from './Footer/Footer';
import B2C from './B2C/B2C';
import HeaderTitle from './HeaderTitle/HeaderTitle';
import PageNavigation from './PageNavigation/PageNavigation';
import NavigationLink from './NavigationLink/NavigationLink';
import Breadcrumbs from './Breadcrumbs/Breadcrumbs';
import MainContainer from './MainContainer/MainContainer';
import PageTitle from './PageTitle/PageTitle';
import FooterNavigation from './FooterNavigation/FooterNavigation';
import FooterLink from './FooterLink/FooterLink';
import PasswordHelpContainer from './PasswordHelp/PasswordHelpContainer';
import PasswordHelp from './PasswordHelp/PasswordHelp';
import PageLevelErrorContainer from './PageLevelErrorContainer/PageLevelErrorContainer';
import CreateNewPassword from './CreateNewPassword/CreateNewPassword';
import DateOfBirth from './DateOfBirth/DateOfBirth';
import TermsAndConditions from './TermsAndConditions/TermsAndConditions';
import ButtonLink from './ButtonLink/ButtonLink';
import Link from './Link/Link';
import Paragraph from './Paragraph/Paragraph';
import Postcode from './Postcode/Postcode';
import InputField from './InputField/InputField';

export default {
    Header: Header,
    Footer: Footer,
    B2C: B2C,
    HeaderTitle: HeaderTitle,
    PageNavigation: PageNavigation,
    NavigationLink: NavigationLink,
    Breadcrumbs: Breadcrumbs,
    MainContainer: MainContainer,
    PageTitle: PageTitle,
    FooterNavigation: FooterNavigation,
    FooterLink: FooterLink,
    PasswordHelpContainer: PasswordHelpContainer,
    PasswordHelp: PasswordHelp,
    PageLevelErrorContainer: PageLevelErrorContainer,
    CreateNewPassword: CreateNewPassword,
    DateOfBirth: DateOfBirth,
    TermsAndConditions: TermsAndConditions,
    ButtonLink: ButtonLink,
    Link: Link,
    Paragraph: Paragraph,
    Postcode: Postcode,
    InputField: InputField
}