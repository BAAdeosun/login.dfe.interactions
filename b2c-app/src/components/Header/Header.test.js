import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';

import components from '..';

it('renders without crashing', () => {
    shallow(<components.Header />);
});

it('renders correctly', () => {
    const tree = renderer
        .create(<components.Header />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('should change state to expand menu when clicking menuToggle', () => {
    const wrapper = mount(<components.Header />);
    expect(wrapper.state().menuExpanded).toBe(false);
    wrapper.find('#menuToggle').simulate('click');
    expect(wrapper.state().menuExpanded).toBe(true);
});