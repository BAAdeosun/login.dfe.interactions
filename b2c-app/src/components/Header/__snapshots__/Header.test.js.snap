// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<header
  className="govuk-header "
  data-module="govuk-header"
  role="banner"
>
  <div
    className="govuk-header__container govuk-width-container ncs-custom"
  >
    <h1
      className="govuk-heading-l govuk-!-margin-top-0 govuk-!-margin-bottom-0"
    >
      <a
        className="govuk-header__link govuk-header__link--service-name"
        href="https://nationalcareers.service.gov.uk/"
      >
        National Careers Service
      </a>
    </h1>
    <button
      aria-controls="navigation"
      aria-label="Show or hide Top Level Navigation"
      className="govuk-header__menu-button govuk-js-header-toggle"
      id="menuToggle"
      onClick={[Function]}
      type="button"
    >
      Menu
    </button>
    <nav>
      <ul
        aria-label="Top Level Navigation"
        className="govuk-header__navigation"
        id="navigation"
      >
        <li
          className="govuk-header__navigation-item"
        >
          <a
            className="govuk-header__link"
            href="https://nationalcareers.service.gov.uk/job-profiles"
          >
            Explore careers
          </a>
        </li>
        <li
          className="govuk-header__navigation-item"
        >
          <a
            className="govuk-header__link"
            href="https://nationalcareers.service.gov.uk/skills-health-check"
          >
            Skills assessment
          </a>
        </li>
        <li
          className="govuk-header__navigation-item"
        >
          <a
            className="govuk-header__link"
            href="https://nationalcareers.service.gov.uk/course-directory"
          >
            Find a course
          </a>
        </li>
        <li
          className="govuk-header__navigation-item"
        >
          <a
            className="govuk-header__link"
            href="https://nationalcareers.service.gov.uk/contact-us"
          >
            Contact us
          </a>
        </li>
        <li
          className="govuk-header__navigation-item"
        >
          <a
            className="govuk-header__link"
            href="https://nationalcareers.service.gov.uk/about-us"
          >
            About us
          </a>
        </li>
        <li
          className="govuk-header__navigation-item"
        >
          <a
            className="govuk-header__link"
            href="https://nationalcareers.service.gov.uk/get-a-job"
          >
            Help to get a job
          </a>
        </li>
      </ul>
    </nav>
  </div>
</header>
`;
